commands/src/main/java/org/wildfly/extras/creaper/commands/deployments/Deploy.java

```diff
@@ -0,0 +1,211 @@
+                mn.add(Constants.SERVER_GROUP, serverGroup);
+                mn.add(Constants.DEPLOYMENT, deploymentName);
+                groupDeploymentAddresses.add(mn);
+            }
+            deployRequest = createDeploymentOperation(content,
+                    groupDeploymentAddresses.toArray(new ModelNode[groupDeploymentAddresses.size()]));
+        } else {
+            deployRequest = createDeploymentOperation(content);
+            ModelNode deployStep = deployRequest.get(Constants.STEPS).add();
+            ModelNode deploymentAddress = new ModelNode();
+            deploymentAddress.add(Constants.DEPLOYMENT, deploymentName);
+            deployStep.set(getEmptyOperation(DEPLOY, deploymentAddress));
+        }
+
+        OperationBuilder builder = new OperationBuilder(deployRequest, autoCloseInputStream);
+        builder.addInputStream(deploymentInputStream);
+        ctx.client.execute(builder.build());
+    }
+
+    private ModelNode createDeploymentOperation(ModelNode content, ModelNode... serverGroupAddresses) {
+        ModelNode composite = getEmptyOperation(Constants.COMPOSITE, new ModelNode().setEmptyList());
+        ModelNode steps = composite.get(Constants.STEPS);
+        ModelNode step1 = steps.add();
+        step1.set(getEmptyOperation(Constants.ADD, new ModelNode().add(Constants.DEPLOYMENT, deploymentName)));
+        step1.get(Constants.CONTENT).add(content);
+        if (serverGroupAddresses != null) {
+            for (ModelNode serverGroup : serverGroupAddresses) {
+                ModelNode sg = steps.add();
+                sg.set(getEmptyOperation(Constants.ADD, serverGroup));
+                sg.get(ENABLED).set(true);
+            }
+        }
+        return composite;
+    }
+
+    private ModelNode getEmptyOperation(String operationName, ModelNode address) {
+        ModelNode op = new ModelNode();
+        op.get(Constants.OP).set(operationName);
+        if (address != null) {
+            op.get(Constants.OP_ADDR).set(address);
+        } else {
+            // Just establish the standard structure; caller can fill in address later
+            op.get(Constants.OP_ADDR);
+        }
+        return op;
+    }
+
+    /**
+     * A single shared instance of this class is passed through the entire builder invocation chain to accumulate
+     * all the options.
+     */
+    private static final class Data {
+
+        private File deploymentFile;
+        private InputStream deploymentInputStream;
+        private final boolean autoCloseInputStream;
+        private final String deploymentName;
+
+        private List<String> serverGroups = null;
+
+        public Data(InputStream deploymentInputStream, String deploymentName, boolean autoCloseInputStream) {
+            this.deploymentInputStream = deploymentInputStream;
+            this.deploymentName = deploymentName;
+            this.autoCloseInputStream = autoCloseInputStream;
+        }
+
+        public Data(File deploymentFile) {
```

Comment:
You are right, good point, as I will redo it to standard builder the whole Data class should disappear.